### ISSUE TITLE
WP_REDIS_SERIALIZER vs WP_REDIS_IGBINARY

### ISSUE THREAD

**Issue by nixsee** [rhubarbgroup/redis-cache#440]
I have Redis Object Cache installed on my WordPress site and have configured Relay on my server. It connects and works fine. In wp-config.php I have the following two lines:

```
define('WP_REDIS_SERIALIZER', Redis::SERIALIZER_IGBINARY);
define('WP_REDIS_IGBINARY', true);
```

And both are recognized in the diagnostics as `2` and `true`, respectively.

It isn't clear to me if I should be using one or the other, or both. When I step through the code with Xdebug, it seems like these two settings are in conflict/competition with each other. If `WP_REDIS_SERIALIZER` is set, it never seems to get to the point where it checks for/uses `WP_REDIS_IGBINARY` - for example, here https://github.com/rhubarbgroup/redis-cache/blob/develop/includes/object-cache.php#L2722

Also, in the [Scaling section of the readme](https://github.com/rhubarbgroup/redis-cache/#scaling), it shows 
`define( 'WP_REDIS_SERIALIZER', 'igbinary' );`

but I was under the impression that you needed to set it as I did above (`Redis::SERIALIZER_IGBINARY`)

So, I'm hoping you could please clarify a bit further here or in the Readme what each one does, why you'd use one over the other, and how to set them. Again, I'm using Relay, but I suppose instructions for both Relay and phpredis would be useful (if there's a difference).

Thanks!

**Comment by tillkruss** [comment]
It looks like the `WP_REDIS_SERIALIZER` is used for PhpRedis and Relay and will set the `Redis::OPT_SERIALIZER` option.

And `WP_REDIS_IGBINARY` will use the `igbinary_*()` methods to compress data.

I think we should remove `WP_REDIS_SERIALIZER`, it's not safe anyhow to use the JSON or MSGPACK options.

**Comment by tillkruss** [comment]
Closed via 4bbcf34da35e60526727209bc5a9cc8235745f85.

**Comment by tillkruss** [comment]
No worries!
